src/interactive_marker_server.cpp

```diff
@@ -388,6 +388,17 @@ bool InteractiveMarkerServer::get( std::string name, visualization_msgs::Interac
   return false;
 }
 
+void InteractiveMarkerServer::getNames(std::vector<std::string>& names)
+{
+  boost::recursive_mutex::scoped_lock lock( mutex_ );
+
+  M_MarkerContext::iterator it;
+  for ( it = marker_contexts_.begin(); it != marker_contexts_.end(); it++ )
+  {
+    names.push_back( it->second.int_marker.name );
```

Comment:
I think this is equivalent to:

```suggestion
    names.push_back( it->first );
```